### PR TITLE
Ocaml packages fix build on darwin

### PR DESCRIPTION
**Pull request by Et7f3**
###### Description of changes

While testing https://github.com/NixOS/nixpkgs/pull/197367 I saw that many other package was broken with the same kind of error so I did a pass to fix the same error in multiple packages.
build log before this PR:
```
error: builder for '/nix/store/vihic2dgbd98lrl0sas2c647c6fz7dsz-ocaml-libvirt-0.6.1.5.drv' failed with exit code 2;
       last 10 log lines:
       > like 'CDEFLMPSUVYZX', is deprecated.
       > Use the equivalent signed form: +C+D+E+F+L+M+P+S+U+V+Y+Z+X-3.
       > Hint: Enabling or disabling a warning by its mnemonic name requires a + or - prefix.
       > Hint: Did you make a spelling mistake when using a mnemonic name?
       > ocamlmklib -o mllibvirt libvirt_c.o libvirt.cmo libvirt_version.cmo -L/nix/store/nv200j8plka6f5knmnsx8wqx7736pq43-libvirt-8.10.0/lib -lvirt
       > ld: file not found: /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation for architecture x86_64
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > make[1]: *** [Makefile:68: mllibvirt.cma] Error 2
       > make[1]: Leaving directory '/private/tmp/nix-build-ocaml-libvirt-0.6.1.5.drv-0/source/libvirt'
       > make: *** [Makefile:31: all] Error 1
       For full logs, run 'nix log /nix/store/vihic2dgbd98lrl0sas2c647c6fz7dsz-ocaml-libvirt-0.6.1.5.drv'.
```

###### Things done

copied from what is done with cairo2

For reviewer:
```
nix build .#ocamlPackages.bjack .#ocamlPackages.ffmpeg .#ocamlPackages.ffmpeg-av .#ocamlPackages.ffmpeg-avcodec .#ocamlPackages.ffmpeg-avdevice .#ocamlPackages.ffmpeg-avfilter .#ocamlPackages.ffmpeg-avutil .#ocamlPackages.ffmpeg-swresample .#ocamlPackages.ffmpeg-swscale .#ocamlPackages.gstreamer .#ocamlPackages.labltk .#ocamlPackages.ocaml_libvirt
```
(might be quicker than nixpkgs-review)

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/gstreamer/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, buildDunePackage, fetchFromGitHub, dune-configurator, pkg-config, glib, gst_all_1 }:
+{ lib, stdenv, buildDunePackage, fetchFromGitHub, dune-configurator, AppKit, Foundation, pkg-config, glib, gst_all_1 }:
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/gstreamer/default.nix:1]: Does I profit of this PR to split deps on each line and sort them ?

File: pkgs/development/ocaml-modules/ffmpeg/ffmpeg-avdevice.nix [hunk 2]
```diff
@@ -11,7 +19,8 @@ buildDunePackage {
   inherit (ffmpeg-base) version src useDune2;
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ dune-configurator ];
+  buildInputs = [ dune-configurator ]
+    ++ lib.optionals stdenv.isDarwin [ AppKit AudioToolbox Cocoa CoreImage ForceFeedback OpenGL VideoToolbox ];
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/ffmpeg/ffmpeg-avdevice.nix:23]: I am unsure what is best split list or not ? ++ on this line or the line before

Comment (wegank) [reply pkgs/development/ocaml-modules/ffmpeg/ffmpeg-avdevice.nix:23]: Both are perfect for me.

File: pkgs/development/ocaml-modules/gstreamer/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/ocaml-modules/gstreamer/default.nix:1]: I would prefer it to be in a separate PR whose name starts with treewide.

**Comment by wegank** [comment]
@ofborg build ocamlPackages.ocaml_libvirt ocamlPackages.labltk ocamlPackages.gstreamer ocamlPackages.ffmpeg-avutil ocamlPackages.bjack ocamlPackages.ffmpeg-avcodec ocamlPackages.ffmpeg-avfilter ocamlPackages.ffmpeg-swscale ocamlPackages.ffmpeg-av ocamlPackages.ffmpeg-avdevice ocamlPackages.ffmpeg-swresample

**Review by wegank — APPROVED** [review]

**Pull request merged**